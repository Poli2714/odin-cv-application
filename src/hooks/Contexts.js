import { createContext } from 'react';

const IndexContext = createContext({
  activeIndex: null,
  setActiveIndex: () => {},
});

const KeyContext = createContext({
  activeKey: {
    eduActiveKey: null,
    expActiveKey: null,
  },
  setActiveKey: () => {},
});

const EduErrMsgContext = createContext({
  showEduErrMessage: false,
  setShowEduErrMessage: () => {},
});

const ExpErrMsgContext = createContext({
  showExpErrMessage: false,
  setShowExpErrMessage: () => {},
});

const PersonalInfoContext = createContext({
  personalInfo: {
    profilePicture: '',
    fullName: '',
    currentJobTitle: '',
    email: '',
    mobile: '',
    city: '',
  },
  setPersonalInfo: () => {},
});

const SummaryContext = createContext({
  summary: '',
  setSummary: () => {},
});

const ExperienceContext = createContext({
  experiences: {
    id: {
      jobTitle: '',
      employer: '',
      jobStartDate: '',
      jobEndDate: '',
      jobDescription: '',
    },
  },
  setExperiences: () => {},
});

const EducationContext = createContext({
  educations: {
    id: {
      degree: '',
      university: '',
      eduStartDate: '',
      eduEndDate: '',
      eduInfo: '',
    },
  },
  setEducations: () => {},
});

export {
  EducationContext,
  EduErrMsgContext,
  ExperienceContext,
  ExpErrMsgContext,
  IndexContext,
  KeyContext,
  PersonalInfoContext,
  SummaryContext,
};
