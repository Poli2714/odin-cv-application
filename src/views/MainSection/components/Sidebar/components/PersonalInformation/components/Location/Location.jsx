import { useContext } from 'react';

import { InputWrapper } from 'src/components/layout';
import { Label, TextInput } from 'src/components/forms';
import { PersonalInfoContext } from 'src/hooks/Contexts';
import styles from './Location.module.css';

function Location() {
  const { personalInfo, setPersonalInfo } = useContext(PersonalInfoContext);
  const { city } = personalInfo;

  const handleChange = e => {
    if (e.target.value.startsWith(' ')) return;
    if (e.target.value.length === 2 && e.target.value.endsWith(' ')) return;
    setPersonalInfo({ ...personalInfo, city: e.target.value });
  };

  return (
    <InputWrapper>
      <Label labelName="City" labelFor="location" />
      <TextInput
        clName={styles.location}
        name="location"
        onChange={handleChange}
        pattern="^.{2,}"
        placeholder="Vancouver"
        value={city}
      />
      {city.length > 0 && city.length < 2 && (
        <div data-testid="warning" className={styles.invalid}></div>
      )}
    </InputWrapper>
  );
}

export default Location;
