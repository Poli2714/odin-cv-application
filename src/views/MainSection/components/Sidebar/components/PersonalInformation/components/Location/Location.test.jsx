import { expect, test, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useState } from 'react';

import Location from './Location';
import { PersonalInfoContext } from 'src/hooks/Contexts';

vi.mock('src/components/layout', () => ({
  InputWrapper: vi.fn(({ children }) => (
    <div data-testid="input-wrapper">{children}</div>
  )),
}));

vi.mock('src/components/forms', () => ({
  Label: vi.fn(() => <label htmlFor="location">Test label</label>),
  TextInput: vi.fn(({ value, onChange }) => (
    <input id="location" type="text" onChange={onChange} value={value} />
  )),
}));

function MockParentComponent() {
  const [personalInfo, setPersonalInfo] = useState({ city: '' });

  return (
    <PersonalInfoContext.Provider value={{ personalInfo, setPersonalInfo }}>
      <Location />
    </PersonalInfoContext.Provider>
  );
}

test('renders Location', () => {
  render(<Location />);

  expect(screen.getByTestId('input-wrapper')).toMatchSnapshot();
});

test('updates city value as a user types', async () => {
  const user = userEvent.setup();

  render(<MockParentComponent />);
  const cityInput = screen.getByLabelText('Test label');

  expect(cityInput).toHaveValue('');
  await user.type(cityInput, 'Baku');
  expect(cityInput).toHaveValue('Baku');
  await user.clear(cityInput);
  await user.type(cityInput, 'Los Angeles');
  expect(cityInput).toHaveValue('Los Angeles');
});
