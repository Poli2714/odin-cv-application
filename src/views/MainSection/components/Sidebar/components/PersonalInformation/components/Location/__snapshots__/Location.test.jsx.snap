// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Location 1`] = `
<div
  data-testid="input-wrapper"
>
  <label
    for="location"
  >
    Test label
  </label>
  <input
    id="location"
    type="text"
    value=""
  />
</div>
`;
